src/pyspiffe/bundle/jwt_bundle/jwt_bundle_set.py

```diff
@@ -2,25 +2,26 @@
 This module manages JwtBundleSet objects.
 """
 
-from typing import Mapping
+import threading
+from typing import Dict, Optional
 from pyspiffe.bundle.jwt_bundle.jwt_bundle import JwtBundle
 from pyspiffe.spiffe_id.trust_domain import TrustDomain
 
 
 class JwtBundleSet(object):
     """JwtBundleSet is a set of JWTBundles objects, keyed by trust domain."""
 
-    def __init__(self, bundles: Mapping[TrustDomain, JwtBundle]) -> None:
-        """Creates a new initialized with the given JWT bundles.
+    def __init__(self, bundles: Dict[TrustDomain, JwtBundle]) -> None:
+        """Creates a new initialized with the given JWT bundles objects keyed by TrustDomain.
```

Comment:
Nit: awkward sentence, maybe "Creates a new set initialized with the given JWT bundle objects keyed by TrustDomain"?